### PR TITLE
Adding emit for adding main location via prompt in controller

### PR DESCRIPTION
**Pull request by subvertnormality**
There was previously no clear way to hook into the adding of a main location via the controller prompt message "Would you like us to use * as your main location to give you relevant local info across the BBC?".

As part of the User Activity Service integration, we need to be able to listen to this event to be able to write the main location to the API.

### CODE REVIEW

**Comment by subvertnormality** [comment]
@tomcollins

File: dist/controller/primary/primary.js [hunk 1]
```diff
@@ -268,6 +268,7 @@ define([
     var setAsPreferredLocation = function() {
       self.preferredLocation.set(location.id, {
         success: function() {
+          $.emit(self.namespace + ':controller:preferred_location', [location]);
           emitLocation();
```
Comment (subvertnormality) [inline dist/controller/primary/primary.js]: We suspect this emitLocation is causing other issues - the main location is being added to the local storage incorrectly.

@mintuz is looking into a fix that will be part of the PR

Comment (tomcollins) [inline dist/controller/primary/primary.js]: This is not the only place that a location may be preferred. It looks like your homepage issue relates to the specific case of a location being preferred through the 'cold start' dialog?

The user location component also allows locations to be preferred by clicking on a star. Locations preferred through this mechanism are not covered by the new event.
https://github.com/bbc/locservices-ui-js/blob/master/src/js/component/user_locations.js#L412

Would is be sufficient to do the following on L271. You could then listen for the existing event on L265 and check for the new property.
```
location.wasSetAsPreferredLocationDueToColdStart = true;
```

Comment (subvertnormality) [inline dist/controller/primary/primary.js]: Hi Tom, that would be sufficient for us - i'll make the amend.

**Comment by mintuz** [comment]
@subvertnormality @tomcollins updated to use a property.

**Comment by tomcollins** [comment]
@subvertnormality @minutz looks good so merging. Thanks.

**Pull request merged**